package-res/resources/web/pentaho/data/filter/AbstractPropertyFilter.js

```diff
@@ -0,0 +1,116 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./AbstractFilter",
+  "../../util/object",
+  "./_apply",
+  "./_toSpec"
+], function(AbstractFilter, O, _apply, toSpec) {
+  "use strict";
+
+  /**
+   * @name AbstractPropertyFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @extends pentaho.data.filter.AbstractFilter
+   * @abstract
+   * @amd pentaho/data/filter/AbstractPropertyFilter
+   *
+   * @classdesc The `AbstractPropertyFilter` class is the abstract base class of
+   * classes that represent a filter (or leaf node of a tree). This class (and its descendants)
+   * extract a given property from within a data table {@link pentaho.data.Table} and
+   * compares its value(s) to the corresponding given candidate value(s) using
+   * some comparison operation.
+   *
+   * @description The `AbstractPropertyFilter` extends the `AbstractFilter`
+   * {@link pentaho.data.filter.AbstractFilter} class to allow creation of
+   * types of filters for a data table {@link pentaho.data.Table}.
+   *
+   * ### Remarks
+   *
+   * The following derived classes are not abstract and can be used directly:
+   *
+   * * {@link pentaho.data.filter.IsEqual}
+   * * {@link pentaho.data.filter.IsIn}
+   *
+   * @param {string} property The name of the property
+   * @param {Object} value The value that belongs to the set
+   *
+   */
+  var AbstractPropertyFilter = AbstractFilter.extend("pentaho.data.filter.AbstractPropertyFilter", /** @lends pentaho.data.filter.AbstractPropertyFilter# */{
+    get type(){
+      /* istanbul ignore next: placeholder getter */
+      return null;
+    },
+
+    constructor: function(property, value) {
+      O.setConst(this, "value", value);
+      O.setConst(this, "property", property);
+    },
+
+    /**
+     * The operation used during the comparison process of the data table's
+     * {@link pentaho.data.Table} candidate value for the given property.
+     *
```

Comment:
This formulation is a bit awkward, do you think you can come up with a clearer alternative?